dashboard/src/api-caller/types.ts

```diff
@@ -6,8 +6,21 @@ export type PaginationResponse<T> = {
   data: T[];
 };
 
+export type TDataSourceConfig_DB = Record<string, never>;
+
+export type TDataSourceConfig_HTTP = {
+  host: string;
+  processing: {
```

Comment:
The configuration for processing needs to be added to the DataSourceConfig type in the API models.
Validation needs to be added in datasource.controller.ts in the validateConfig function